src/indexer.cc

```diff
@@ -1611,6 +1616,36 @@ void OnIndexReference(CXClientData client_data, const CXIdxEntityRefInfo* ref) {
 FileContents::FileContents(const std::string& path, const std::string& content)
     : path(path), content(content) {}
 
+FileContentsWithOffsets::FileContentsWithOffsets() : line_offsets_{0} {}
+
+FileContentsWithOffsets::FileContentsWithOffsets(std::string s) {
+  contents_ = s;
+  line_offsets_.push_back(0);
+  for (size_t i = 0; i < s.size(); i++)
+    if (s[i] == '\r' || s[i] == '\n') {
+      if (s[i] == '\r' && i + 1 < s.size() && s[i + 1] == '\n')
+        i++;
+      line_offsets_.push_back(i + 1);
+    }
+}
+
+optional<int> FileContentsWithOffsets::ToOffset(Position p) const {
+  if (0 < p.line && size_t(p.line) <= line_offsets_.size()) {
```

Comment:
p.line > 0

Reply:
I use this order for resembling the math `a < b <= c` as much as possible...